sdk/keyvault/keyvault-keys/src/lro/delete/operation.ts

```diff
@@ -2,110 +2,127 @@
-      state.isCompleted = true;
+  /**
+   * @summary Reaches to the service and updates the delete key's poll operation.
+   * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+   */
+  public async update(
+    options: {
+      abortSignal?: AbortSignalLike;
+      fireProgress?: (state: DeleteKeyPollOperationState) => void;
+    } = {}
+  ): Promise<DeleteKeyPollOperation> {
+    const state = this.state;
+    const { name } = state;
+
+    if (options.abortSignal) {
+      this.requestOptions.abortSignal = options.abortSignal;
     }
-  }
 
-  if (!state.isCompleted) {
-    try {
-      state.result = await client.getDeletedKey(name, { requestOptions });
-      state.isCompleted = true;
-    } catch (error) {
-      if (error.statusCode === 403) {
-        // At this point, the resource exists but the user doesn't have access to it.
-        state.isCompleted = true;
-      } else if (error.statusCode !== 404) {
-        state.error = error;
+    if (!state.isStarted) {
+      const deletedKey = await this.deleteKey(name, this.requestOptions);
+      state.isStarted = true;
+      state.result = deletedKey;
+      if (!deletedKey.properties.recoveryId) {
         state.isCompleted = true;
       }
     }
-  }
-
-  return makeDeleteKeyPollOperation(state);
-}
-
-/**
- * @summary Reaches to the service and cancels the key's operation, also updating the key's poll operation
- * @param [options] The optional parameters, which is only an abortSignal from @azure/abort-controller
- */
-async function cancel(this: DeleteKeyPollOperation): Promise<DeleteKeyPollOperation> {
-  throw new Error("Canceling the deletion of a key is not supported.");
-}
 
-/**
- * @summary Serializes the create key's poll operation
- */
-function toString(this: DeleteKeyPollOperation): string {
-  return JSON.stringify({
-    state: this.state
-  });
-}
+    if (!state.isCompleted) {
+      try {
+        state.result = await this.getDeletedKey(name, {
+          requestOptions: this.requestOptions
+        });
+        state.isCompleted = true;
+      } catch (error) {
+        if (error.statusCode === 403) {
+          // At this point, the resource exists but the user doesn't have access to it.
+          state.isCompleted = true;
+        } else if (error.statusCode !== 404) {
+          state.error = error;
+          state.isCompleted = true;
+        }
+      }
+    }
 
-/**
- * @summary Builds a create key's poll operation
- * @param [state] A poll operation's state, in case the new one is intended to follow up where the previous one was left.
- */
-export function makeDeleteKeyPollOperation(
-  state: DeleteKeyPollOperationState
-): DeleteKeyPollOperation {
-  return {
-    state: {
-      ...state
-    },
-    update,
-    cancel,
-    toString
-  };
+    return new DeleteKeyPollOperation(state, this.vaultUrl, this.client, this.requestOptions);
```

Comment:
I'll update it to return `this`! Thank you!

We talked about this over private chat and the reason why update expects a return is because of: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-lro/src/poller.ts#L316

The current contract of the poller allows for stateless workflows.

I think we made decisions like this in core-lro because of the flexibility it gave us. So, no rush on making that contract change, in my opinion!

Reply:
We could do something like: `this.operation = update() || this.operation`, for example. Even more flexibility! (though the argument that we might not need this much flexibility is valid. I just feel more comfortable making things less restrictive 😅)